karapace/protobuf/enum_element.py

```diff
@@ -0,0 +1,61 @@
+# Ported from square/wire:
+# wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/EnumElement.kt
+from itertools import chain
+from karapace.protobuf.compare_result import CompareResult, Modification
+from karapace.protobuf.compare_type_storage import CompareTypes
+from karapace.protobuf.enum_constant_element import EnumConstantElement
+from karapace.protobuf.location import Location
+from karapace.protobuf.type_element import TypeElement
+from karapace.protobuf.utils import append_documentation, append_indented
+
+
+class EnumElement(TypeElement):
+    def __init__(
+        self, location: Location, name: str, documentation: str = "", options: list = None, constants: list = None
+    ) -> None:
+        # Enums do not allow nested type declarations.
+        super().__init__(location, name, documentation, options or [], [])
+        self.constants = constants or []
```

Comment:
This class and class hierarchy was ported from square/wire. I prefer to keep this class and hierarchy unchanged. 
And because @dataclass is not so flexible I see no acceptable way to use @dataclass construction for class EnumElement without breaking of the class hierarchy.